packages/fxa-settings/src/components/DropDownAvatarMenu/index.tsx

```diff
@@ -0,0 +1,126 @@
+  mutation destroySession($input: DestroySessionInput!) {
+    destroySession(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const DropDownAvatarMenu = () => {
+  const { displayName, primaryEmail } = useAccount();
+  const [isRevealed, setRevealed] = useState(false);
+  const toggleRevealed = () => setRevealed(!isRevealed);
+  const avatarMenuInsideRef = useClickOutsideEffect<HTMLDivElement>(
+    setRevealed
+  );
+  useEscKeydownEffect(setRevealed);
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const dropDownId = 'drop-down-avatar-menu';
+
+  // TODO: FXA-2450
+  const [destroySession, { data, error }] = useMutation(
+    DESTROY_SESSION_MUTATION,
+    {
+      onError: (error) => {
+        revealAlertBar();
+        sentryMetrics.captureException(error);
+      },
+    }
+  );
+
+  const signOut = () => {
+    destroySession({
+      variables: { input: {} },
+    });
+  };
+
+  if (data) {
+    clearSignedInAccountUid();
+    window.location.assign(`${window.location.origin}/signin`);
+    return null;
+  }
+
+  return (
+    <>
+      {alertBarRevealed && error && (
+        <AlertBar onDismiss={hideAlertBar}>
+          <p data-testid="sign-out-error">Error text TBD. {error.message}</p>
+        </AlertBar>
+      )}
+      <div className="relative" ref={avatarMenuInsideRef}>
+        <button
+          onClick={toggleRevealed}
+          data-testid="drop-down-avatar-menu-toggle"
+          title="Firefox Account Menu"
+          aria-expanded={isRevealed}
+          aria-controls={dropDownId}
+          className="rounded-full border-2 border-transparent hover:border-purple-500 focus:border-purple-500 focus:outline-none active:border-purple-700 transition-standard"
+        >
+          <Avatar className="w-10 rounded-full" />
+        </button>
+        {isRevealed && (
+          <div
+            id={dropDownId}
+            data-testid={dropDownId}
+            className="drop-down-menu -left-52"
+          >
+            <div className="flex flex-wrap">
+              <div className="flex w-full p-4">
+                <div className="mr-3 flex flex-1">
+                  <Avatar className="w-10" />
+                </div>
+                <div className="flex flex-4 truncate">
+                  <p className="leading-5 max-w-full">
+                    <span className="text-grey-400 text-xs truncate">
+                      Signed in as
+                    </span>
+                    <span
+                      className="font-bold block max-w-full truncate"
+                      data-testid="drop-down-name-or-email"
+                    >
+                      {displayName || primaryEmail.email}
+                    </span>
+                  </p>
+                </div>
+              </div>
+              <div className="w-full">
+                <div className="bg-gradient-to-r from-blue-500 via-pink-700 to-yellow-500 h-px" />
+                <div className="p-6">
```

Comment:
Nitpick: something like `px-4 py-5` might be a little closer to the padding in Figma

Reply:
Hmm, I'm not sure, `p-6` is 24px all the way around. If anything, I could bump the left padding _up_ to a `7`.

<img width="244" alt="image" src="https://user-images.githubusercontent.com/13018240/91460848-f2b07d80-e84d-11ea-9932-73c06d322d78.png">
<img width="159" alt="image" src="https://user-images.githubusercontent.com/13018240/91460866-f80dc800-e84d-11ea-944c-eb99cfe883e0.png">

Reply:
I'll leave this one as-is, we can address it in a design review if Phil/Jess prefer less padding here.

Reply:
for posterity

![Artboard](https://user-images.githubusercontent.com/6392049/91462337-11b40d00-e858-11ea-8bff-022125f42356.png)